classes/user_enrolment_repository.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace local_augmented_teacher;
+
+class user_enrolment_repository
+{
+    /**
+     * Get user enrolments by course ids
+     * @param array $course_ids
+     * @return array
+     * @throws \dml_exception
+     */
+    public function get_by_course_ids(array $course_ids): array {
+        global $DB;
```

Comment:
Please add $DB as a param in the constructur.

Like this:
public function __construct(moodle_database $db) {
   $this->db = $db;
}